system-tests/lib/cre/features/consensus/v1/consensus.go

```diff
@@ -0,0 +1,234 @@
+package v1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	chainselectors "github.com/smartcontractkit/chain-selectors"
+
+	capabilitiespb "github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
+	kcr "github.com/smartcontractkit/chainlink-evm/gethwrappers/keystone/generated/capabilities_registry_1_1_0"
+	jobv1 "github.com/smartcontractkit/chainlink-protos/job-distributor/v1/job"
+
+	"github.com/smartcontractkit/chainlink-deployments-framework/offchain/jd"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	keystone_changeset "github.com/smartcontractkit/chainlink/deployment/keystone/changeset"
+	ks_contracts_op "github.com/smartcontractkit/chainlink/deployment/keystone/changeset/operations/contracts"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/config"
+
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/contracts"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/don/jobs"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/don/jobs/ocr"
+	"github.com/smartcontractkit/chainlink/system-tests/lib/cre/features/consensus"
+)
+
+const flag = cre.ConsensusCapability
+
+type Consensus struct{}
+
+func (c *Consensus) Flag() cre.CapabilityFlag {
+	return flag
+}
+
+func (c *Consensus) PreEnvStartup(
+	ctx context.Context,
+	testLogger zerolog.Logger,
+	registryChainSelector uint64,
```

Comment:
@Tofel I would love to see a more structured API here

something like:

PreEnvStartup(context.Context, logger, environment, topology)

In my mind: topology is the description of DONs and Nodes; environment is everything else (JD, CapReg, WorkflowReg, Chains)

i.e. we would move the registryChainSelector to the environment, and the GatewayJobConfigs to the topology

Reply:
done